syntax = "proto3";

package profile;


message UserID {
    int32 user_id = 1;
}

message ProfileData {
    string phone_number = 1;
    string image = 2;
    string city = 3;
    int32 age = 4;
    int32 height = 5;
    int32 weight = 6;
    string skill_level = 7;
    string gender = 8;
    string first_name = 9;
    string name = 10;
    string username = 11;
}


message CreateProfileData {
    string phone_number = 1;
    int32 id = 2;
    string first_name = 3;
    string name = 4;
}


message SuccessProfile {

    string message = 1;
}



message ProfileInfo {
    string username = 1;
    string photo = 2;
    int32 age = 3;

}


message RequestMediaProfile {

    int32 user_id = 1;
    int32 video_id = 2;

    int32 score = 3;
    double calories = 4;

}


message GetAdvencedMedia {
    int32 user_id = 1;
    int32 video_id = 2;

}

message AdvencedMediaData {

    int32 score = 1;
    float calories = 2;

}
 
message SmallRequstProfile {

    string username = 1;
    string photo = 2;

    
}
    

service Profile {
    rpc DataProfile (UserID) returns (ProfileData);
    rpc CreateProfile (CreateProfileData) returns (SuccessProfile);
    rpc ProfileInfoTournament (UserID) returns (ProfileInfo);
    rpc MediaProfile (RequestMediaProfile) returns (SuccessProfile);
    rpc GetAdvencedMediaProfile(GetAdvencedMedia) returns (AdvencedMediaData);
    rpc SmallProfileInfo (UserID) returns (SmallRequstProfile);
    rpc DeleteAccount (UserID) returns (SuccessProfile);

}

